modules/csvparser/csvparser.c

```diff
@@ -167,7 +167,7 @@ csv_parser_new(GlobalConfig *cfg)
   self->super.super.free_fn = csv_parser_free;
   self->super.super.clone = csv_parser_clone;
   self->super.process = csv_parser_process;
-  self->formatted_key = g_string_sized_new(32);
+  self->formatted_key = scratch_buffers2_alloc();
   csv_scanner_options_set_delimiters(&self->options, " ");
```

Comment:
This is not ok, as we need to allocate the key so that it is thread safe. This would be reused by all csv_parser_process() calls, not to mention that it would be freed. Remember that you can't keep a reference.

This should be allocated right in the process() method call and probably not even stored in "self".